rudder-core/src/main/scala/com/normation/rudder/services/workflows/WorkflowService.scala

```diff
@@ -144,6 +146,14 @@ trait WorkflowLevelService {
   def getForDirective  (actor: EventActor, change: DirectiveChangeRequest  ): Box[WorkflowService]
   def getForNodeGroup  (actor: EventActor, change: NodeGroupChangeRequest  ): Box[WorkflowService]
   def getForGlobalParam(actor: EventActor, change: GlobalParamChangeRequest): Box[WorkflowService]
+
+  /*
+   * These method allow to get change request impacting a rule/directive/etc.
+   * Used to display information on them on corresponding update screens.
+   */
+  def getByDirective(id : DirectiveId, onlyPending:Boolean) : Box[Vector[ChangeRequest]]
```

Comment:
Just a question here, why a Vector and not an IndexedSeq ?   (or even a Seq/List? )

But don't change it, it's just for my knowledge

Reply:
I just moved (or copy/pasted) interfaces that where provided elsewhere. 

Not sure for the rationnal, but the idea should have been along the line: be precise about what you give to let consumer adpat their behaviors.